src/site/topics/testing/unit-test-collections.rst

```diff
@@ -158,6 +158,206 @@ Common Setup Code
 Singleton Case Stack Tests
 ==========================
 
+* Below are example unit tests for all methods within the ``ArrayStack`` class when it has a single element within it
+* Notice the inclusion of the new class field called ``preState``
+* This ``preState`` will effectively be a duplicate of ``classUnderTest`` that can be used to check that a method had no side effect
+* Consider ``peek_singleton_unchanged`` for an example
+
+    * Calling ``peek`` should not have any side effect; it should not mutate the object in any way
+    * To verify this, one can assert equality between ``preState`` and ``classUnderTest`` after calling ``peek`` on ``classUnderTest``
+
+
+.. code-block:: java
+    :linenos:
+    :emphasize-lines: 2,7,47,49
+
+    private ArrayStack<Integer> classUnderTest;
+    private ArrayStack<Integer> preState;
+
+    @BeforeEach
+    void createStack() {
+        classUnderTest = new ArrayStack<>();
+        preState = new ArrayStack<>();
+    }
+
+    // Empty tests excluded here
+
+    @Test
+    void push_singleton_returnsTrue() {
+        classUnderTest.push(10);
+        assertTrue(classUnderTest.push(11));
```

Comment:
despite not being used, `preState` should be updated here as well -- really hammers home (imo) why `@BeforeEach` and `@Nested` are simplifications and useful.

Reply:
I'm conflicted on this. I agree that it certainly motivates the `@BeforeEach` value, but it's also not used at all in any test but one. It's in the many tests where it's used again, which is the next section. 

I'll think on it. It's kinda' a funny, peculiar situation.

Reply:
I ended up adding it with a note. I think the note helps makes it make more sense why it's a class field and always set to be equal to `classUnderTest`.